### PR TITLE
Add checksum for py-virtualenv 20.16.4

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
@spackbot fix style

**Review by tldahlgren — COMMENTED** [review]
Confirmed sha256.

**Comment by iarspider** [comment]
Can I ask for  #33047 to be merged now, since this PR kind of depends on it?

File: var/spack/repos/builtin/packages/py-virtualenv/package.py [hunk 1]
```diff
@@ -22,20 +23,28 @@ class PyVirtualenv(PythonPackage):
...
-    depends_on("python@2.7:2.8,3.5:", type=("build", "run"), when="@20.10.0:")
+    depends_on("python@2.7:2.8,3.5:", when="@20.10.0:", type=("build", "run"))
+    depends_on("python@3.6:", when="@20.16.4:", type=("build", "run"))
 
     # not just build-time, requires pkg_resources
     depends_on("py-setuptools@40.6.3:", type=("build", "run"))
-    depends_on("py-setuptools@41.00.03:", type=("build", "run"), when="@20.10.0:")
-    depends_on("py-setuptools-scm@2:", type=("build", "run"), when="@20.10.0:")
+    depends_on("py-setuptools@41.00.03:", when="@20.10.0:", type=("build", "run"))
+    depends_on("py-setuptools-scm@2:", when="@20.10.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-virtualenv/package.py]: These need to be updated from `pyproject.toml`:
```toml
requires = [
    "setuptools>=59.6",
    "setuptools-scm>=6.4.2",
]
```
Also, setuptools-scm prob is only needed at build time.

File: var/spack/repos/builtin/packages/py-virtualenv/package.py [hunk 2]
```diff
@@ -22,20 +23,28 @@ class PyVirtualenv(PythonPackage):
...
+    depends_on("py-filelock@3.4.1:3", when="@20.16.4:", type=("build", "run"))
+    depends_on("py-platformdirs@2:2", when="@20.10.0:", type=("build", "run"))
+    depends_on("py-platformdirs@2.4:2", when="@20.16.4:", type=("build", "run"))
+    depends_on("py-six@1.9.0:1", when="@20.10.0", type=("build", "run"))
+    depends_on("py-importlib-metadata@0.12:", when="@20.10.0: ^python@:3.7", type=("build", "run"))
+    depends_on("py-importlib-resources@1:", when="@20.10.0: ^python@:3.6", type=("build", "run"))
+    depends_on(
+        "py-importlib-metadata@4.8.3:", when="@20.10.0: ^python@:3.7", type=("build", "run")
+    )
+    depends_on("py-importlib-resources@5.4:", when="@20.10.0: ^python@:3.6", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-virtualenv/package.py]: I wouldn't expect these `when` versions to be identical, did you mean `@20.16.4:`?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@alalazo this PR was failing audit tests (missing dep version) but still auto-merged. Any idea what happened here?

**Comment by adamjstewart** [comment]
Btw, if #33047 ever passes CI, this issue will be solved in develop.

**Comment by adamjstewart** [comment]
Are our audit tests not required for CI to pass?

**Comment by alalazo** [comment]
They should be, let me check

**Pull request merged**